pep-9999.rst

```diff
@@ -0,0 +1,883 @@
+
+ - restricting the upload of packages to the public Pypi index to the packaes
+   using only FOSS (Free and open source software) licenses.
+
+
+Motivation
+==========
+
+Software is licensed and providing accurate licensing information to Python
+packages users is an important matter.  Today, there are multiple places where
+license is documented in package metadata and there are limitations to what can
+be documented. This is often leading to confusion or a lack of clarity both for
+package authors and package users.
+
+Several package authors have expressed difficulty and/or frustrations with the
+possibilities to express licensing in package metadata. This also applies to
+Liux distribution packagers. This has triggered several license-related
+discussions and issues and in particular:
+
+- https://github.com/pypa/warehouse/issues/2996
+- https://github.com/pypa/interoperability-peps/issues/46
+- https://github.com/pypa/packaging-problems/issues/41
+- https://github.com/pypa/wheel/issues/138
+- https://github.com/pombredanne/spdx-pypi-pep/issues/1
+
+On average, Python packages tend to have more ambiguous or missing license
+information than other common application package formats (such as npm, Maven or
+Gem) as can be seen in the statistics [#cdstats]_ page of the ClearlyDefined
+[#cd]_ project that cover all packages from Pypi, Maven, npm and Rubygems.
+ClearlyDefined is an open source project to help improve clarity of other open
+source projecs that is incubating at the OSI (Open Source Initiative) [#osi]_.
+
+
+Rationale
+=========
+
+A mini survey of existing license metadata definitions in use in Python today
+and documented in several other system/distro and application package formats is
+provided as an informative Appendix in this PEP.
+
+There are a few takeaways from this survey:
+
+- Most package formats use a single `license` field.
+
+- Many modern package formats use some form of license expression syntax to
+  optionally combine more than one license identifiers together. SPDX and
+  SPDX-like syntaxes are the most popular in use.
+
+- SPDX license ids are becoming a de-facto way to reference common licenses
+  everywhere, whether or not a license expression syntax is used. But they often
+  need to be supplemented with extra license ids or conventions to accept extra
+  or generic licenses such as "Proprietary" or "Public domain" not tracked by
+  SPDX.
+
+- Several package format support documenting both a license expression and
+  the paths of the corresponding files that contain the license text. Most free
+  and open source software licenses require to include their full text in a
+  distribution.
+
+These considerations have guided the design and updates proposed in this PEP.
+
+Reusing the existing `license` field with license expressions will provide a an
+intuitive and more structured way to express the license of a distribution using
+a well defined syntax and well known license ids.
```

Comment:
Note: license list is moving target. New identifiers are added.

Reply:
It is a moving target, but a slow moving one. Since I propose only a soft validation that issues a warning for unknown ids, the adoption of a new license ids that is not yet known from a tool would only trigger a warning.

Reply:
@phadej we are tracking the discussion on this in https://github.com/pombredanne/spdx-pypi-pep/issues/6 if you mind chiming in there.